### ISSUE TITLE
suggest to flatten simple nested matches

### ISSUE THREAD

**Issue by oli-obk** [Manishearth/rust-clippy#1253]
We should add a lint that suggests to change

```rust
match foo {
    SomeEnum::Variant(boo) => match boo {
        OtherEnum::Variant2(x) => ...,
        OtherEnum::Variant3(y) => ...,
    },
    SomeEnum::Variant4 => ...,
}
```

into

```rust
match foo {
    SomeEnum::Variant(OtherEnum::Variant2(x)) => ...,
    SomeEnum::Variant(OtherEnum::Variant3(y)) => ...,
    SomeEnum::Variant4 => ...,
}
```

finding such situations is pretty easy, building a good suggestion might be a little tricky though
